### PR TITLE
Fesc ssfr and jwst magnitude

### PR DESCRIPTION
**Pull request by qyx268**
a couple of changes included including the anchoring of the two fesc models. but most importantly, this PR includes the calculation of JWST photometries.

### CODE REVIEW

File: src/core/reionization.c [hunk 1]
```diff
@@ -104,12 +104,12 @@ void update_galaxy_fesc_vals(galaxy_t* gal, double new_stars, int snapshot)
 #endif
       }
       break;
-    case 6: // specific star formation rate (1 / Myr)
+    case 6: // specific star formation rate (10/ Gyr)
       if ((gal->Sfr > 0.0) && (gal->StellarMass > 0.0)) {
-        fesc *= pow(gal->Sfr / gal->StellarMass / run_globals.units.UnitTime_in_s * SEC_PER_MEGAYEAR,
+        fesc *= pow(gal->Sfr / gal->StellarMass / run_globals.units.UnitTime_in_s * SEC_PER_MEGAYEAR / 100,
```
Comment (EMventura) [inline src/core/reionization.c:109]: That makes sense, just out of curiosity why are we doing this? Is it to match some particular formalism?

File: input/photometric_tables/NIRCam_Wide/F444W_wavelength.bin
Comment (EMventura) [inline input/photometric_tables/NIRCam_Wide/F444W_wavelength.bin:1]: I haven't personally opened these files but I trust these are the correct ones :D

**Review by EMventura — COMMENTED** [review]
All of this makes sense to me except for two things that are not fully clear to me, if you could me explain the logic I am happy to approve the merge :)

I haven't run the code, do you want me to test the code as well?

File: src/core/magnitudes.c [hunk 2]
```diff
@@ -177,8 +214,41 @@ void init_templates_mini(mag_params_t* miniSpectra,
...
+    }
+
+    jwst_transmission_splined = (double*)malloc(n_splined*sizeof(double));
+    jwst_lambda_splined = (double*)malloc(n_splined*sizeof(double));
+
+    iwave_offset = 0;
+    for (iband=0; iband<N_JWST; iband++){
+        for (iwave=0; iwave<jwst_number[iband]; iwave++){
+            if (iwave==0)
+                jwst_lambda_splined[iwave+iwave_offset] = spectra[iS].waves[iwave] * (1.+ redshifts[nAgeStep] +1e-4);
```
[the remark this answers is not in the document]
Comment (qyx268) [reply src/core/magnitudes.c:231]: yes

File: src/core/magnitudes.c [hunk 3]
```diff
@@ -206,7 +276,7 @@ void init_templates_mini(mag_params_t* miniSpectra,
     init_templates_special(spectra + iS, tBC, 1);
 #if USE_MINI_HALOS
     init_templates_rawIII(spectraIII + iS, fNameIII);
-    init_filters(spectraIII + iS, betaBands, nBeta, restBands, nRest, NULL, NULL, NULL, 0, 1. + redshifts[iS]);
+    init_filters(spectraIII + iS, betaBands, nBeta, restBands, nRest, jwst_transmission_splined, jwst_lambda_splined, jwst_number, N_JWST, redshifts[nAgeStep]);
```
[the remark this answers is not in the document]
Comment (qyx268) [reply src/core/magnitudes.c:279]: it was wrong.

[the remark this answers is not in the document]
Comment (qyx268) [reply src/core/magnitudes.c:279]: but we were not doing obs bands (i.e. all the NULL), so it was fine...

File: src/core/reionization.c (hunk 1 above)
Comment (qyx268) [reply src/core/reionization.c:109]: i updated this based on Rosdahl MNRAS 515, 2386–2414 (2022)

File: src/core/magnitudes.c [hunk 4]
```diff
@@ -177,8 +214,41 @@ void init_templates_mini(mag_params_t* miniSpectra,
...
+    }
+
     // Initialise filters
-    init_filters(spectra + iS, betaBands, nBeta, restBands, nRest, NULL, NULL, NULL, 0, 1. + redshifts[iS]);
+    init_filters(spectra + iS, betaBands, nBeta, restBands, nRest, jwst_transmission_splined, jwst_lambda_splined, jwst_number, N_JWST, redshifts[nAgeStep]);
+    for (iwave=0; iwave<MAGS_N_BANDS; iwave++){
+        //mlog("iwave = %d: spectra.centreWave=%.1f",MLOG_MESG, iwave, spectra[iS].centreWaves[iwave]);
+        miniSpectra->allcentreWaves[iS][iwave] = spectra[iS].centreWaves[iwave];
+    }
     if (spectra[iS].nFlux != MAGS_N_BANDS) {
```
[the remark this answers is not in the document]
Comment (qyx268) [reply src/core/magnitudes.c:316]: updated. i'm doing some tests

**Comment by qyx268** [comment]
@all-contributors please add @jakebolam for infrastructure, tests and code

**Comment by qyx268** [comment]
https://github.com/all-contributors please add @smutch @qyx268 @EMventura @BradGreig @s-balu @gbpoole @pgeil @yqiuu

**Comment by qyx268** [comment]
@all-contributors please add @smutch @qyx268 @EMventura @BradGreig @s-balu @gbpoole @pgeil @yqiuu @daviesje for code

File: src/core/magnitudes.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (EMventura) [reply src/core/magnitudes.c]: Ok!

**Comment by qyx268** [comment]
https://github.com/all-contributors please add  @qyx268  for code, maintenance, review

**Comment by qyx268** [comment]
@all-contributors please add @qyx268 for code, maintenance, review

**Comment by qyx268** [comment]
@all-contributors please add @EMventura  for code, maintenance, review

**Comment by qyx268** [comment]
@all-contributors please add @EMventura for code, maintenance, review
@all-contributors please add @BradGreig for code
@all-contributors please add @gbpoole for code
@all-contributors please add @pgeil for code
@all-contributors please add @yqiuu for code
@all-contributors please add @daviesje for code

File: README.md [hunk 5]
```diff
@@ -1,90 +1,43 @@
...
 
-* **Simon Mutch** (Lead author; *The University of Melbourne*)
-* Paul Geil (*The University of Melbourne*)
-* Yuxiang Qin (*The University of Melbourne*)
-* Hansik Kim (*The University of Melbourne*)
-* Greg Poole (*Astronomy Data and Compute Services*)
-* Yisheng Qiu (*The University of Melbourne*)
-* Brad Greig (*The University of Melbourne*)
-* Emanuele Maria Ventura (*The University of Melbourne*)
+- **AGN Model:**  Qin et al. (2017). **Dark-ages reionization and galaxy formation simulation - X. The small contribution of quasars to reionization.** *Monthly Notices of the Royal Astronomical Society*, 472 (2): 2009–2027. [DOI: 10.1093/mnras/stx1909](https://doi.org/10.1093/mnras/stx1909).
```
Comment (EMventura) [inline README.md:29]: Maybe adding Qiu+19 for the supernova model? (As you prefer though)

**Comment by qyx268** [comment]
@all-contributors please add @EMventura for code, review

**Comment by qyx268** [comment]
@all-contributors please add @EMventura for code, review

**Review by EMventura — COMMENTED** [review]
Thanks for this! New JWST and HST magnitudes is indeed a cool feature and thanks also for the comments in the code that indeed helped me to follow what is going on. 
For me this is a thumbs up, just wanted to double check with you if tested this version with both MINI_HALOS = Off and ON. If yes and the code runs I approve the merge

**Comment by qyx268** [comment]
@all-contributors please add @gbpoole for code

**Comment by qyx268** [comment]
@all-contributors please add @pgeil for code

File: README.md (hunk 5 above)
Comment (qyx268) [reply README.md:29]: will do

**Comment by qyx268** [comment]
@all-contributors please add @yqiuu for code

**Comment by qyx268** [comment]
@all-contributors please add @daviesje for code

**Comment by qyx268** [comment]
am I missing any contributors?

**Comment by EMventura** [comment]
> am I missing any contributors?

Balu?

**Comment by qyx268** [comment]
@all-contributors please add @s-balu  for code, review

**Comment by s-balu** [comment]
Hi @qyx268 the lighcone looks alright when I view it.  But I have made a new one with only the L210_AUG:

**Comment by qyx268** [comment]
it's okay. probably there was some cache...

**Comment by pgeil** [comment]
Hi. Can someone please remove me from this. Thanks,PaulSent from my telephoneOn 22 Nov 2024, at 21:12, Yuxiang Qin ***@***.***> wrote:﻿
it's okay. probably there was some cache...

**Comment by qyx268** [comment]
> Hi. Can someone please remove me from this. Thanks,PaulSent from my telephoneOn 22 Nov 2024, at 21:12, Yuxiang Qin ***@***.***> wrote:﻿ it's okay. probably there was some cache... —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

done :)

**Comment by qyx268** [comment]
I think the magnitudes calculations are correct. Here are 2 proofs:
1. the intrinsic 1500 UV LF with and without dust 
image
2. the observed SEDs in the 8 JWST and 2 HST filters (showing 10 random galaxies in each redshift using lines) together with the intrinsic 1500 magnitude (the points; after K correction). You see they overlap. 
image

**Comment by smutch** [comment]
I think when Paul asked to be removed, he probably just meant that he didn't want to keep receiving the emails for this thread. I'd like to make sure he is appropriately credited though as a contributor to the code. :)

**Comment by qyx268** [comment]
so you reckon i can put him back on the readme doc? i added allcontributors because apparently github doesnt show contributors automatically on the right-side bar...

**Comment by qyx268** [comment]
@smutch btw, do you know if Hank has a github account?

**Comment by smutch** [comment]
> @smutch btw, do you know if Hank has a github account?

Not that I know of I'm afraid.

**Review by EMventura — APPROVED** [review]
Everything looks good to me! Approved!

**Pull request merged**